cupy/_core/core.pyx

```diff
@@ -995,6 +995,9 @@ cdef class _ndarray_base:
            :meth:`numpy.ndarray.max`
 
         """
+        if hasattr(self, '__cupy_override_reduction_kernel__'):
```

Comment:
I would only like to have this check in the `ReductionKernel` machinery. Do you think it should be possible to do that? thanks!

Reply:
If we remove this check, `cupy._core._routines_statistics._ndarray_max` tries CUB and cuTENSOR before falling back to `cupy.max`.
So this check should be done before reaching `cupy.max.__call__`, I wonder where this check should ideally be placed.
https://github.com/shino16/cupy/blob/main/cupy/_core/_routines_statistics.pyx#L25-L43

Reply:
That's a good call. Maybe we should rewrite those checks to make it feasible.
Like having an attribute in the ndarray that's overridden by derived classes.
`array.support_cub_for_routines`

Reply:
Now `DistributedArray` overrides all the methods of `ndarray` including max/min/sum/prod, so those hooks in `cupy._core.core.ndarray` are removed. Still your idea sounds reasonable.